src/agent/coverage/src/block/linux.rs

```diff
@@ -130,60 +150,46 @@ impl Recorder {
     }
 
     fn on_module_load(&mut self, tracee: &mut Tracee, image: &ModuleImage) -> Result<()> {
-        log::info!("module load: {}", image.path().display());
-
-        let blocks = find_module_blocks(image.path())?;
+        log::info!("module load: {}", image.path());
 
-        log::debug!("found {} blocks", blocks.len());
+        // Fetch disassembled module info via cache.
+        let info = self.modules.fetch(image.path())?;
 
-        if blocks.is_empty() {
-            // This almost certainly means the binary was stripped of symbols.
-            log::warn!("no blocks for module, not setting breakpoints");
-            return Ok(());
+        if info.is_none() {
+            anyhow::bail!("unable to fetch info for module: {}", image.path());
         }
 
-        let new = self.coverage.add_module(image, &blocks)?;
+        let info = info.unwrap();
 
-        if new {
-            for b in blocks {
-                let va = image.offset_to_va(b.offset);
-                log::trace!("set breakpoint: {:x} (~{})", va, tracee.pid);
-                self.breakpoints.set(tracee, va)?;
+        // Collect blocks allowed by the symbol filter.
+        let mut allowed_blocks = vec![];
+
+        for symbol in info.module.symbols.iter() {
```

Comment:
i believe you can achieve the same without allocating the vector first with flat_map

``` rust
let allowed_blocks =
      info.module.symbols.iter()
      .filter(|symbol| {
          self.symbol_filter
              .is_allowed(&info.module1.path, &symbol.name)
      })
      .flat_map(|symbol| info.blocks.range(symbol.range()))
      .collect();
```

Reply:
True, though I find this style harder to read, so I think I'm going to leave as-is.